pkg/util/util.go

```diff
@@ -323,3 +323,19 @@ func GenerateUniqueHashForWorkloadImageTag(namespace string) string {
 	h.Write([]byte(namespace))
 	return base64.URLEncoding.EncodeToString(h.Sum(nil))[0:5]
 }
+
+// GenerateRandomRouteName returns a random, trimmed route name based on the Component name based on the following criteria
+// 1. Under 30 characters
+// 2. Contains 4 random characters
+func GenerateRandomRouteName(componentName string) string {
+	routeName := componentName
+	if len(componentName) > 25 {
+		routeName = componentName[0:24]
```

Comment:
actually, should this be `componentName[0:25]` since that will result in 25 chars?  I think you had it set to that in a previous commit.